### PR TITLE
fix(register): #MA-1029 fix create and remove event function sql

### PR DESCRIPTION
**Pull request by BastienBenard**
## Describe your changes
Fogot else part in sql function.

## Checklist tests
Put a departure and a comment on a call.

## Issue ticket number and link
[MA-1029](https://jira.support-ent.fr/browse/MA-1029)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: presences/src/main/resources/sql/056-MA-1029-fix-add-and-remove-event-alert.sql [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            IF existingAbsenceWithAlert.id IS NULL AND NOT presences.absence_exclude_alert(NEW, register.structure_id) THEN
+                -- Create alert
+                EXECUTE presences.create_alert(NEW.id, 'ABSENCE', NEW.student_id, register.structure_id, register.start_date);
+            END IF;
+        WHEN 2 THEN -- Lateness creation
+            IF NOT presences.lateness_exclude_alert(NEW, register.structure_id) THEN -- If we have no exclude condition
+                -- Create alert
+                EXECUTE presences.create_alert(NEW.id, 'LATENESS', NEW.student_id, register.structure_id, register.start_date);
+            END IF;
+        ELSE
```
Comment (BastienBenard) [inline presences/src/main/resources/sql/056-MA-1029-fix-add-and-remove-event-alert.sql:23]: Ajout seulement de ce Else voir presences/src/main/resources/sql/048-MA-966-create-trigger-make-alerts-configurable.sql

File: presences/src/main/resources/sql/056-MA-1029-fix-add-and-remove-event-alert.sql [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            EXECUTE presences.delete_alert(OLD.id, 'ABSENCE', OLD.student_id, register.structure_id);
+            -- If we have other absence in the same time with no alert, we need a alert for this absence
+            -- The 2 events are necessarily on the same structure, no need to recalculate the register
+            IF eventNeedAlert.id IS NOT NULL AND NOT presences.absence_exclude_alert(eventNeedAlert, register.structure_id) THEN
+                SELECT * FROM presences.register WHERE id = eventNeedAlert.register_id LIMIT 1 INTO register;
+                EXECUTE presences.create_alert(eventNeedAlert.id, 'ABSENCE', OLD.student_id, register.structure_id, register.start_date);
+            END IF;
+        WHEN 2 THEN -- Lateness creation
+        EXECUTE presences.delete_alert(OLD.id, 'LATENESS', OLD.student_id, register.structure_id);
+        ELSE
```
Comment (BastienBenard) [inline presences/src/main/resources/sql/056-MA-1029-fix-add-and-remove-event-alert.sql:51]: Ajout seulement de ce Else voir presences/src/main/resources/sql/048-MA-966-create-trigger-make-alerts-configurable.sql

**Review by sjollois — APPROVED** [review]

File: presences/src/main/resources/sql/056-MA-1029-fix-add-and-remove-event-alert.sql (hunk 1 above)
Comment (sjollois) [reply presences/src/main/resources/sql/056-MA-1029-fix-add-and-remove-event-alert.sql:23]: cela faisait quoi sans le ELSE ?

**Review by flmartineau — APPROVED** [review]

File: presences/src/main/resources/sql/056-MA-1029-fix-add-and-remove-event-alert.sql (hunk 1 above)
Comment (BastienBenard) [reply presences/src/main/resources/sql/056-MA-1029-fix-add-and-remove-event-alert.sql:23]: Les evenement peuvent avoir 4 type:
1 absences 2 retards 3 départ 4 observation
Pour les départ et les retard cela provoquais l'erreur suivante:
```
org.postgresql.util.PSQLException: ERROR: case not found
  Hint: CASE statement is missing ELSE part.
  Where: PL/pgSQL function presences.add_event_alert() line 8 at CASE
  ```

**Pull request merged**